llvm/lib/CodeGen/BranchFolding.h

```diff
@@ -50,11 +50,11 @@ class TargetRegisterInfo;
     class MergePotentialsElt {
       unsigned Hash;
       MachineBasicBlock *Block;
-      DebugLoc BranchDebugLoc;
+      MDNode *BranchDebugLoc;
```

Comment:
The change to use `MDNode` here is a special case; due to some quirk of the allocation of these objects in BranchFolding, we can't use a type that contains a SmallVector here, which DebugLoc does with the CMake flag enabled.